File: frontend/src/types/entityEvent.ts
import type { LongId } from '@/types/common'

export interface EntityEventType {
  id: LongId
  tenantId?: LongId
  categoryCode: string
  categoryName: string
  categoryPolarity?: string // POSITIVE/NEGATIVE/NEUTRAL
  typeCode: string
  typeName: string
  hasScore?: boolean
  hasSeverity?: boolean
  severityLevels?: string | null
  icon?: string | null
  color?: string | null
  applicableSubjects?: string | null
  isSystem?: boolean
  isEnabled?: boolean | number
  sortOrder?: number
}

export interface EntityEvent {
  id: LongId
  subjectType: string
  subjectId: LongId
  subjectName: string | null
  eventCategory: string
  eventType: string
  eventLabel: string | null
  payload: string | null  // JSON
  sourceModule: string | null
  sourceRefType: string | null
  sourceRefId: LongId | null
  tags: string | null  // JSON array
  createdBy: number | null
  createdByName: string | null
  occurredAt: string
}

export interface EntityEventStats {
  subjectType: string
  subjectId: LongId
  totalCount: number
  categoryBreakdown: Record<string, number>
  recentCount: number
}

export interface CreateEntityEventTypeCommand {
  categoryCode: string
  categoryName: string
  /** 分类极性 POSITIVE/NEGATIVE/NEUTRAL */
  categoryPolarity?: string
  typeCode: string
  typeName: string
  /** 创建时是否启用 (兼容旧调用方传 number) */
  isEnabled?: boolean | number
  hasScore?: boolean
  hasSeverity?: boolean
  severityLevels?: string | null
  icon?: string | null
  color?: string | null
  applicableSubjects?: string | null
  sortOrder?: number
}

export interface UpdateEntityEventTypeCommand extends Omit<Partial<CreateEntityEventTypeCommand>, 'isEnabled'> {
  isEnabled?: boolean | number
}

File: backend/src/main/java/com/school/management/domain/event/model/EntityEvent.java
package com.school.management.domain.event.model;

import lombok.Builder;
import lombok.Getter;

import java.time.LocalDateTime;
import java.util.List;

/**
 * 实体事件聚合根
 */
@Getter
@Builder
public class EntityEvent {

    private Long id;
    private Long tenantId;
    private String subjectType;
    private Long subjectId;
    private String subjectName;
    private String eventCategory;
    private String eventType;
    private String eventLabel;
    private String payload;           // JSON
    private String sourceModule;
    private String sourceRefType;
    private Long sourceRefId;
    private String tags;              // JSON数组
    private Long createdBy;
    private String createdByName;
    private LocalDateTime occurredAt;
    private LocalDateTime createdAt;

    // 关联主体（非持久化，仅内存）
    private List<EntityEventRelation> relations;

    /**
     * 工厂方法：创建新事件
     */
    public static EntityEvent create(String subjectType, Long subjectId, String subjectName,
                                     String eventCategory, String eventType, String eventLabel,
                                     String payload, String sourceModule, String sourceRefType,
                                     Long sourceRefId, String tags, Long createdBy, String createdByName) {
        return EntityEvent.builder()
                .subjectType(subjectType)
                .subjectId(subjectId)
                .subjectName(subjectName)
                .eventCategory(eventCategory)
                .eventType(eventType)
                .eventLabel(eventLabel)
                .payload(payload)
                .sourceModule(sourceModule)
                .sourceRefType(sourceRefType)
                .sourceRefId(sourceRefId)
                .tags(tags)
                .createdBy(createdBy)
                .createdByName(createdByName)
                .occurredAt(LocalDateTime.now())
                .tenantId(0L)
                .build();
    }
}
